pkg/services/ngalert/store/alert_rule.go

```diff
@@ -64,20 +64,30 @@ func getAlertRuleByUID(sess *sqlstore.DBSession, alertRuleUID string, orgID int6
 }
 
 // DeleteAlertRuleByUID is a handler for deleting an alert rule.
-func (st DBstore) DeleteAlertRuleByUID(ctx context.Context, orgID int64, ruleUID string) error {
+func (st DBstore) DeleteAlertRulesByUID(ctx context.Context, orgID int64, ruleUID ...string) error {
 	return st.SQLStore.WithTransactionalDbSession(ctx, func(sess *sqlstore.DBSession) error {
-		_, err := sess.Exec("DELETE FROM alert_rule WHERE org_id = ? AND uid = ?", orgID, ruleUID)
```

Comment:
As @joeblubaugh said, could we perhaps take the opportunity to improve the way we handle arguments for these IN clauses?

I think something like the below should work (you can choose to use or not use the "affected rows") - but you get the gist of it.

```golang
rows, err := sess.Table("alert_rule").Where("org_id = ?", orgID).In("uid", ruleUID).Delete(ngmodels.AlertRule{})
if err != nil {
	return err
}
st.Logger.Debug("deleted alert rules", "count", rows)

rows, err = sess.Table("alert_rule_version").Where("rule_org_id = ?", orgID).In("rule_uid", ruleUID).Delete(ngmodels.AlertRuleVersion{})
if err != nil {
	return err
}
st.Logger.Debug("deleted alert rule versions", "count", rows)

rows, err = sess.Table("alert_instance").Where("rule_org_id = ?", orgID).In("rule_uid", ruleUID).Delete(ngmodels.AlertInstance{})
if err != nil {
	return err
}
st.Logger.Debug("deleted alert instances", "count", rows)
```

Reply:
Oh, that's nice xorm-foo. I am personally inclined in using xorm methods but could not find that technique in docs